pkg/tfbridge/provider.go

```diff
@@ -1350,6 +1357,7 @@ func (p *Provider) Delete(ctx context.Context, req *pulumirpc.DeleteRequest) (*p
 	// Create a new destroy diff.
 	diff := p.tf.NewDestroyDiff(ctx, res.TFName, shim.TimeoutOptions{
 		TimeoutOverrides: newTimeoutOverrides(shim.TimeoutDelete, req.Timeout),
+		ResourceTimeout:  res.TF.Timeouts(),
```

Comment:
Why do you not need to decode them here but had to decode the timeouts in Update?

Reply:
Yeah it looks really weird. I copied what's in Create to what's in Update however in Delete there's no data to decode. I'm wondering if these two forms are equivalent actually.

Reply:
I can take a deeper dive to figure out if we can just use res.TF.Timeouts() everywhere.

Reply:
https://github.com/pulumi/pulumi-terraform-bridge/issues/1753 I've posted findings here, there is a slight difference and I'm inclined to keep it as is with keeping the mystery for later.

Reply:
I'm happy with that, just wanted to make sure it is intentional